src/code/z_lib.c

```diff
@@ -334,7 +348,7 @@ void (*sInitChainHandlers[])(u8* ptr, InitChainEntry* ichain) = {
     IChain_Apply_Vec3f, IChain_Apply_Vec3fdiv1000, IChain_Apply_Vec3s,
 };
 
-void Actor_ProcessInitChain(Actor* actor, InitChainEntry* ichain) {
+void Actor_ProcessInitChain(void* actor, InitChainEntry* ichain) {
```

Comment:
If the arg is changing, it probably shouldn't be prefixed with `Actor_` anymore. Maybe a new name like `InitChain_Process` would be more suitable at that point. Although, this would be a pretty big change that isn't directly related to the PR title. Maybe this can be done separately?

Reply:
btw initchain macros use `offsetof(Actor, x)`

Reply:
I don't think a name change is necessary, my idea here is more just to avoid having to define Actor for something that isn't actually type safe internally if it means I can avoid including another header. 

Angelo mentioned you could just use `struct Actor* actor` which I'd be ok with if everyone else is.

Reply:
There's nothing specific to actors here right? I'd be in favor of removing the Actor dependency and calling it `IChain_Process(void*, InitChainEntry*)` or something

Reply:
As mentioned before, the current implementation of the InitChain macros assume that all members belong to the type Actor. We could change this (and I probably argued for it back when), but there also isn't much benefit in doing so because the InitChain system is not actually space or speed efficient; you're simply better off assigning each member directly.

Reply:
Since it's assuming the type is Actor and there's no tangible benefit to making it generic, I think the arg should be actor (or more specifically, `struct Actor*` is fine)